ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -1441,6 +1441,9 @@ def collect(self, device, unused):
             results['hypervisor_details'][hypervisor_id] = hypervisor_detail_id
 
         results['servers'] = yield self.api_call(nova.servers, 'servers')
+        # Retry as single in case where we don't use administrator account.
+        if not results['servers']:
```

Comment:
technically this will also waste some time by doing the query twice if there just are no servers.. We could add a way for api_call to optionally raise the permission error and catch it here.. but honestly it probably doesn't really matter.. the extra api call should be harmless and fast.